system-x/services/aws/kinesis/src/main/java/software/tnb/aws/kinesis/validation/KinesisValidation.java

```diff
@@ -44,7 +43,11 @@ public void sendMessage(String stream, String message, String partitionKey) {
         client.putRecord(b -> b.streamName(stream).partitionKey(partitionKey).data(SdkBytes.fromUtf8String(message)));
     }
 
-    public List<Record> getRecords(String streamName) {
+    public GetRecordsResponse getRecords(String streamName) {
```

Comment:
in what other data from `GetRecordsResponse` are you interested in?

Reply:
In general my idea was to extend the `getRecords` method to be able to return more records.
One option was just to add `getRecords(String streamName, int maxNumberOfRecords)`, however this would still limit number of returned records to 10000.
To be able to return more records user has to use iterator from previous `getRecords` request, which is returned in GetRecordsResponse as `nextShardIterator()`.
Personally I don't need such functionality for now, but I felt it would be good to have it. Alternatively the next iterator could be stored in `KinesisValidation`, though that would limit the API flexibility. Or the iterator handling can be dropped completely.

Reply:
makes sense, thanks